context_test.go

```diff
@@ -1,12 +1,12 @@
-package weave
+package weave_test
```

Comment:
Fine, but unrelated change to the PR.
Why did you pull this out btw?

I am fine with both approaches, more curious here

Reply:
Isolating namespaces is nice and also requires better testability from the original code.
Since I have added my test as weave_test I have decided to convert this one too